lib/rule-tester/rule-tester.js

```diff
@@ -578,6 +582,10 @@ class RuleTester {
 
             if (hasOwnProperty(item, "options")) {
                 assert(Array.isArray(item.options), "options must be an array");
+                if (!rule.schema) {
```

Comment:
I think we shouldn't log the warning if `item.options` is an empty array, as that's basically the same as if no options had been passed.